### ISSUE TITLE
Linker error when using device_vector::resize()

### ISSUE THREAD

**Issue by eligao** [thrust/thrust#792]
I came into an issue when using device_vector on VS2013 and CUDA 7.5.18 on Windows 10.
### The most simple way of reproducing this issue:
- On VS2013, create a new`CUDA 7.5 runtime` project
- replace everything in `kernel.cu` with:

```
#include <thrust/device_vector.h>
int main(void)
{
    thrust::device_vector<int> vec;
    vec.resize(1);
    return 0;
}
```
- In the project properties, under `CUDA C/C++`, set Code Generation as `compute_35,sm_35` and Generate Relocatable Device Code as `-rdc=true`
- Compile the project, and we can get a linker error:

```
1>CudaDemo2.device-link.obj : error LNK2001: unresolved external symbol __fatbinwrap_66_tmpxft_00000888_00000000_17_cuda_device_runtime_compute_52_cpp1_ii_8b1a5d37
1>D:\foo\CudaDemo2\x64\Debug\CudaDemo2.exe : fatal error LNK1120: 1  unresolved externals
```
### When all conditions below are met, the linker error occurs.
1. use Generate Relocatable Device Code (it works well without `-rdc=true`)
2. choose computing capability >=3.5 (I tried 3.5, 5.0 and 5.2 with no luck, but `compute_30,sm_30` works well)
3. invoke `thrust::device_vector::resize()` (not sure whether there are other functions with the same issue, but the similar function `thrust::device_vector::reserve()` works fine)

I tried with the lastest commit  3e6b4ff but with no luck.

**Comment by eligao** [comment]
This is a sample solution with this issue:
https://www.dropbox.com/s/vogvz1p4pmq7oik/CudaDemo2.zip?dl=0

**Comment by chipweinberger** [comment]
Has this been solved?

If I disable rdc, what would you recommend I do with all my device functions to still get them to compile?

**Comment by chipweinberger** [comment]
I should add, this problem also happens with thrust::unique

**Comment by eligao** [comment]
It's been a long time since I posted this issue. I don't remember solving this bug, I probably just avoided using this function and used others as workaround. 
What is your current version of cuda and thrust? Did you try cuda8?

**Comment by brycelelbach** [comment]
Is this still a problem?

**Comment by brycelelbach** [comment]
#885 is probably a duplicate of this.

I think this is a compiler problem, not a Thrust problem. Could you try reproducing the issue with a CUDA program that doesn't use Thrust?

I've filed a bug internally for this (2004557).

**Comment by brycelelbach** [comment]
Fix given in #885.
